src/ApiException.php

```diff
@@ -78,6 +78,49 @@ public function getStatus()
         return $this->status;
     }
 
+    // Returns null if metadata does not contain error info, or return containsErrorInfo() array
+    // if the metadata does contain error info.
+    public static function decodeMetadataErrorInfo($metadata)
```

Comment:
Two things here:
1. Since this is only being called internally and surfaced by other public methods, it should be private
2. As mentioned by Noah, all methods should use PHP doc! For example (although I am not sure if this is accurate):

```suggestion
    /**
     * Returns null if metadata does not contain error info, or return containsErrorInfo() array
     * if the metadata does contain error info.
     *
     * @param array $metadata
     * @return array|null
     *./
    private static function decodeMetadataErrorInfo(array $metadata)
```

Reply:
Makes sense - updated!